### PR TITLE
Remove unused ability to override KonnectivityImage

### PR DESCRIPTION
**Pull request by enxebre**
Initially we allow overrding defaultKonnectivityImage so we could pin an image while let product builders inject their own. Konnectiviy image comes from payload now.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/konnectivity/params.go [hunk 1]
```diff
@@ -60,9 +60,7 @@ func NewKonnectivityParams(hcp *hyperv1.HostedControlPlane, images map[string]st
 	if _, ok := images["apiserver-network-proxy"]; ok {
 		p.Image = images["apiserver-network-proxy"]
 	}
-	if _, ok := hcp.Annotations[hyperv1.KonnectivityAgentImageAnnotation]; ok {
```
Comment (enxebre) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/konnectivity/params.go:63]: @relyt0925 @hasueki to validate you are fine dropping this

**Comment by enxebre** [comment]
/hold
to confirm https://github.com/openshift/hypershift/pull/3101#discussion_r1360945400

**Review by hasueki — CHANGES_REQUESTED** [review]
/hold IBM Cloud is still using this annotation for 4.10 clusters

File: control-plane-operator/main.go [hunk 2]
```diff
@@ -341,8 +331,6 @@ func NewStartCommand() *cobra.Command {
 		componentImages := map[string]string{
 			util.AvailabilityProberImageName: availabilityProberImage,
 			"hosted-cluster-config-operator": hostedClusterConfigOperatorImage,
-			"konnectivity-server":            konnectivityServerImage,
```
[the remark this answers is not in the document]
Comment (enxebre) [reply control-plane-operator/main.go:344]: Seems konnectivity-agent is fetched in the guest cluster agent `NewKonnectivityParams`

**Pull request closed without merging**